Wootric/WootricSDK-iOS
# Event queue

Add support to handle events. This would bring full support of `Targeted Sampling` to iOS. The goal is to be able to call the SDK with an event name and let the server decide if the end user combined with the event name and properties make the user eligible to be surveyed. The implementation adds an `NSOperationQueue` to handle the events being called in a `FIFO` way.

## Changes
- Add custom implementation of `NSOperation`
- Refactor code to handle multiple calls
- Update and add tests

## Test

- Set up `Targeted Sampling` in one account
- Call survey with the event name and properties to trigger the survey

```objective-c
  NSString *clientID = @"YOUR_CLIENT_ID";
  NSString *accountToken = @"YOUR_ACCOUNT_TOKEN";

  [Wootric configureWithClientID:clientID accountToken:accountToken];
  [Wootric setEndUserEmail:@"END_USER_EMAIL"];
  [Wootric setEndUserCreatedAt:@1234567890];
  [Wootric setEndUserProperties:@{ @"property1": @"value1", @"property2": @"value2"}];


  [Wootric showSurveyInViewController:self event:@"Button Click"];
```

Try calling this multiple times with different values, the SDK should handle the different calls separately (even try different calls with different `endUserEmail`, `endUserProperties` and `eventName`).

Once the server returns `true`, all subsequent calls should be cancelled and the survey should be shown with the correct values.

- Test this on both iPhone and iPad
